runelite-client/src/main/java/net/runelite/client/plugins/deathindicator/DeathIndicatorPlugin.java

```diff
@@ -0,0 +1,355 @@
+import net.runelite.client.ui.overlay.infobox.Timer;
+import net.runelite.client.ui.overlay.worldmap.WorldMapPointManager;
+
+@PluginDescriptor(
+	name = "Death Indicator"
+)
+@Slf4j
+public class DeathIndicatorPlugin extends Plugin
+{
+	public static final BufferedImage WORLDMAP_ARROW;
+	public static final BufferedImage BONES;
+
+	private WorldPoint deathLocation;
+	private Timer deathTimer;
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private DeathIndicatorConfig config;
+
+	@Inject
+	private DeathIndicatorPlugin plugin;
+
+	@Inject
+	private WorldMapPointManager worldMapPointManager;
+
+	@Inject
+	private InfoBoxManager infoBoxManager;
+
+	@Inject
+	private ConfigManager configManager;
+
+	static
+	{
+		try
+		{
+			synchronized (ImageIO.class)
+			{
+				WORLDMAP_ARROW = ImageIO.read(DeathIndicatorPlugin.class.getResourceAsStream("worldmap_arrow.png"));
+				BONES = ImageIO.read(DeathIndicatorPlugin.class.getResourceAsStream("bones.png"));
+			}
+		}
+		catch (IOException e)
+		{
+			throw new RuntimeException(e);
+		}
+	}
+
+	@Provides
+	DeathIndicatorConfig deathIndicatorConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(DeathIndicatorConfig.class);
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		client.clearHintArrow();
+
+		if (config.worldMapPointSet())
+		{
+			worldMapPointManager.removeIf(DeathWorldMapPoint.class::isInstance);
+			config.worldMapPointSet(false);
+		}
+
+		if (deathTimer != null)
+		{
+			infoBoxManager.removeInfoBox(deathTimer);
+		}
+	}
+
+	@Subscribe
+	public void onActorDeath(ActorDeath event)
+	{
+		if (client.isInInstancedRegion())
+		{
+			return;
+		}
+
+		Actor actor = event.getActor();
+
+		if (actor instanceof NPC || actor != client.getLocalPlayer())
+		{
+			return;
+		}
+
+		config.hasDied(true);
+
+		config.hasRespawned(false);
+
+		config.deathLocationX(actor.getWorldLocation().getX());
+		config.deathLocationY(actor.getWorldLocation().getY());
+		config.deathLocationPlane(actor.getWorldLocation().getPlane());
+
+		deathLocation = new WorldPoint(config.deathLocationX(), config.deathLocationY(), config.deathLocationPlane());
+
+		config.deathWorld(client.getWorld());
+
+		if (config.showDeathHintArrow())
+		{
+			client.setHintArrow(deathLocation);
+		}
+
+		if (config.showDeathOnWorldMap())
+		{
+			addMapPoint(deathLocation);
+		}
+
+		if (deathTimer != null)
+		{
+			infoBoxManager.removeInfoBox(deathTimer);
+		}
+
+		if (config.showDeathInfoBox())
+		{
+			config.timeOfDeath(Instant.now().toString());
+			deathTimer = new Timer(60, ChronoUnit.MINUTES, BONES, plugin);
+			deathTimer.setTooltip("Died on world: " + config.deathWorld());
+			infoBoxManager.addInfoBox(deathTimer);
+		}
+	}
+
+	@Subscribe
+	public void onChatMessage(ChatMessage event)
+	{
+		if (event.getMessage().equals("Oh dear, you are dead!"))
+		{
+			config.hasRespawned(true);
+		}
+
+		if ((config.hasDied()) && (!config.timeOfDeath().equals("")))
```

Comment:
To move one statement out of the heavily nested if statements, you could do:
```java
if (!config.hasDied() && config.timeofDeath().equals(""))
{
    return;
}
```

Reply:
I like that idea, implemented this change to remove one of those nested loops, it looks much better now. Thanks!